### ISSUE TITLE
Add `named` bool to DirectedEdge

### ISSUE THREAD

**Issue by ericrwolfe** [valhalla/baldr#328]
Currently there's no lightweight way of knowing whether or not a directed edge has any street names. 

Was tossing the idea around with @kevinkreiser the other day about adding a bool to indicate whether an edge is "named" or not. 

Such a bool would allow for costing profiles (particularly pedestrian) that prefer named edges and would lead to less confusing/ambiguous narratives.

**Comment by kevinkreiser** [comment]
Essentially we were talking about using a bit on the edge that let us know if the edge had a name or not

**Comment by dnesbitt61** [comment]
Seems reasonable, though not sure I agree preferring named edges for pedestrian. I think there are lots of unnamed pathways through parks and such that create a good route but maybe are harder to describe.

**Comment by ericrwolfe** [comment]
Agreed. We talked about the park example and it sounded like you have plans to work on that. More examples of associating paths with nearby polygons would be walkways that parallel named streets or other geographic features such as waterways or coastlines.

The named edges bool would serve as a great stopgap until landmark/polygon information is supported. Idea was to use a slight (0.9f) preference for named edges that would help in confusing cases such as when an unnamed walkway/sidewalk parallels a named street (I've run into this many times in testing).

**Comment by ericrwolfe** [comment]
Excellent, thanks for the addition!
